src/main/java/io/vertx/codegen/generators/dataobjecthelper/DataObjectHelperGen.java

```diff
@@ -99,48 +97,49 @@ private void genToJson(String visibility, boolean inheritConverter, DataObjectMo
+          if (dataObject != null) {
+            if (dataObject.isSerializable()) {
+              String m;
+              MapperInfo mapperInfo = dataObject.getSerializer();
+              String match;
+              switch (mapperInfo.getKind()) {
+                case SELF:
+                  m = "";
+                  match = ".toJson()";
+                  break;
+                case STATIC_METHOD:
+                  m = mapperInfo.getQualifiedName() + "." + String.join(".", mapperInfo.getSelectors()) + "(";
+                  match = ")";
+                  break;
+                default:
+                  throw new UnsupportedOperationException();
               }
-              break;
-            case ENUM:
-              genPropToJson("", ".name()", prop, writer);
-              break;
-            case JSON_OBJECT:
-            case JSON_ARRAY:
-            case OBJECT:
-              genPropToJson("", "", prop, writer);
-              break;
-            case DATA_OBJECT:
-              DataObjectTypeInfo dataObjectType = ((DataObjectTypeInfo)prop.getType());
-              if (dataObjectType.isSerializable()) {
-                String m;
-                MapperInfo mapperInfo = dataObjectType.getSerializer();
-                String match;
-                switch (mapperInfo.getKind()) {
-                  case SELF:
-                    m = "";
-                    match = ".toJson()";
-                    break;
-                  case STATIC_METHOD:
-                    m = mapperInfo.getQualifiedName() + "." + String.join(".", mapperInfo.getSelectors()) + "(";
-                    match = ")";
-                    break;
-                  default:
-                    throw new UnsupportedOperationException();
+              genPropToJson(m, match, prop, writer);
+            } else {
+              return;
+            }
+          } else {
+            switch (propKind) {
+              case API:
+                if (prop.getType().getName().equals("io.vertx.core.buffer.Buffer")) {
+                  genPropToJson("java.util.Base64.getEncoder().encodeToString(", ".getBytes())", prop, writer);
                 }
-                genPropToJson(m, match, prop, writer);
-              } else {
-                return;
-              }
-              break;
-            case OTHER:
-              if (prop.getType().getName().equals(Instant.class.getName())) {
-                genPropToJson("DateTimeFormatter.ISO_INSTANT.format(", ")", prop, writer);
-              }
-              break;
+                break;
+              case ENUM:
+                genPropToJson("", ".name()", prop, writer);
+                break;
+              case JSON_OBJECT:
+              case JSON_ARRAY:
+              case OBJECT:
+                genPropToJson("", "", prop, writer);
+                break;
+              case OTHER:
+                if (prop.getType().getName().equals(Instant.class.getName())) {
```

Comment:
Same as above, can we find a way to remove it registering `Instant` as `ClassTypeInfo` with `DataObjectInfo`?

Reply:
in another PR.